src/main/java/org/xbill/DNS/LookupResult.java

```diff
@@ -0,0 +1,61 @@
+package org.xbill.DNS;
+
+import static java.util.Objects.hash;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.xbill.DNS.exceptions.AdditionalDetail;
+
+/** LookupResult instances holds the result of a successful lookup operation. */
+public class LookupResult {
+  private final List<Record> records;
+  private final AdditionalDetail additionalDetail;
+
+  /**
+   * Construct an instance with the provided records.
+   *
+   * @param records a list of records to return, or null if there was no response
+   * @param additionalDetail additional detail on this response, such as the reason for records
+   *     being null.
+   */
+  public LookupResult(List<Record> records, AdditionalDetail additionalDetail) {
+    this.records = Collections.unmodifiableList(new ArrayList<>(records));
+    if (records.isEmpty() && additionalDetail == null) {
+      throw new IllegalArgumentException(
+          "Empty records can not be combined with null additionalDetail");
+    }
+    this.additionalDetail = additionalDetail;
+  }
+
+  /**
+   * An unmodifiable list of records that this instance wraps
+   *
+   * @return an unmodifiable List of Record instances.
+   */
+  public List<Record> get() {
+    return records;
+  }
+
+  public AdditionalDetail getAdditionalDetail() {
+    return additionalDetail;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
You can use [Lombok](https://projectlombok.org/features/EqualsAndHashCode) for this: add `@EqualsAndHashCode` to the class.

Reply:
Thank you for reminding me. I have encountered lombok at the margins but never really engaged with it. This seems very useful. It seems if you add `lombok.addLombokGeneratedAnnotation = true` to a `lombok.config` in the toplevel directory it will exclude generated code from the code coverage calculation as well. I'll add that.

Reply:
I don't think I can ever code Java without Lombok again :-)

Please make sure the option doesn't introduce a runtime dependency on Lombok via the `@lombok.Generated` annotation.

Reply:
Some googling seems to suggest that annotations that byte code parser encountering annotations that are not on the classpath will ignore them, at least after this bug was fixed: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6322301

Reply:
Guava has a bunch of transitive dependencies from annotation libraries that shouldn't be necessary. Not quite sure where the issue lies exactly, Maven, Java, an IDE, whatever, but they need to be there that stuff doesn't fail. See e.g. the discussions in google/guava#3302.

Reply:
Having taken a brief look at the referenced issues I don't think that these are comparable, but I will verify this locally.

Reply:
I may have picked up the wrong issues in Guava, but all the stuff in their dependency chain that comes along with Guava (failureaccess, listenablefuture, jsr305, errorprone, checker-qual, etc.) is all just for static analysis tools but somehow needed at runtime, at least in some situations (that aren't obvious).